### PR TITLE
Added a way to reuse index.html for all other pages

### PR DESCRIPTION
**Pull request by chiragsanghvi**
This PR fixes issue https://github.com/informatics-isi-edu/chaise/issues/266 by consolidating `index.html.in` templates into 1 for reuse. across different apps 

https://dev.isrd.isi.edu/~chirag/chaise/recordset/#1/isa:dataset

### CODE REVIEW

**Comment by chiragsanghvi** [comment]
The build is passing. Can I merge it.

File: Makefile [hunk 1]
```diff
@@ -577,40 +577,82 @@ distclean: clean
 html: $(HTML)
 
 # Rules to attach JavaScript and CSS assets to the head
-search/index.html: search/index.html.in .make-asset-block .make-template-block
+search/index.html: .make-asset-block .make-template-block
```
Comment (robes) [inline Makefile]: `search/search.html.in` should be a dependency, right?

File: Makefile [hunk 2]
```diff
@@ -577,40 +577,82 @@ distclean: clean
...
 	sed -e '/%ASSETS%/ {' -e 'r .make-asset-block' -e 'd' -e '}' \
 		-e '/%TEMPLATES%/ {' -e 'r .make-template-block' -e 'd' -e '}' \
-		search/index.html.in > search/index.html
+		search/search.html.in > search/index.html
 
-login/index.html: login/index.html.in .make-asset-block
+login/index.html: .make-asset-block
```
Comment (robes) [inline Makefile]: and `index.html.in` should be a dependency here and throughout the rest of the rules, right?

File: Makefile [hunk 3]
```diff
@@ -577,40 +577,82 @@ distclean: clean
...
-detailed/index.html: detailed/index.html.in .make-detailed-asset-block .make-detailed-template-block
+		-e 's/%APPNAME%//' \
+		-e 's/%PAGE%/Chaise :: Login/' \
+		-e 's/%NAVBAR%//' \
+		-e 's/%FOOTER%//' \
+		-e 's/%TEMPLATES%//' \
+		-e '/%CONTENT%/ {' -e 'r login/login.html.in' -e 'd' -e '}' \
+		index.html.in > login/index.html
+
+viewer/index.html: .make-viewer-asset-block
```
Comment (robes) [inline Makefile]: these blocks all repeat... seems like they could be collapsed into a single pattern based rule?

**Comment by chiragsanghvi** [comment]
@robes Should I merge this. I have made the changes as you requested.

**Comment by robes** [comment]
Hmmm, I don't know. This seems more complicated. Like I commented before, I was thinking that a "pattern rule" might simplify this. Meaning you just need one rule something like:

```
%/index.html: index.html.in .make-%-asset-template
    sed -e '/%ASSETS%/ {' -e 'r .make-$*-asset-block' -e 'd' -e '}' \
        -e '/%TEMPLATES%/ {' -e 'r .make-$*-template-block' -e 'd' -e '}' \
        index.html.in > $*/index.html
```

This is what I had in mind with my earlier suggestion. I personally would find this change to the defined sequence `generateHtml` hard to follow and maintain.

**Comment by hongsudt** [comment]
@chiragsanghvi have you addressed @robes's coments? If so, then please resolve the conflict and merge the code.

**Comment by hongsudt** [comment]
@chiragsanghvi can you resolve this with @robes and finish it?

**Comment by RFSH** [comment]
Since this PR was created we changed the structure of our apps and therefore this PR is not valid anymore. I am going to close this.
I am going to keep the issue that is connected to this PR open, so we can discuss later whether it's worth doing this with the new structure.
